torchgeo/datasets/splits.py

```diff
@@ -0,0 +1,359 @@
+    Returns
+        A list of the subset datasets.
+
+    .. versionadded:: 0.5
+    """
+    new_indexes = [
+        Index(interleaved=False, properties=Property(dimension=3)) for _ in rois
+    ]
+
+    for i, roi in enumerate(rois):
+        if any(roi.intersects(x) and (roi & x).area > 0 for x in rois[i + 1 :]):
+            raise ValueError("ROIs in input rois can't overlap.")
+
+        j = 0
+        for hit in dataset.index.intersection(tuple(roi), objects=True):
+            box = BoundingBox(*hit.bounds)
+            new_box = box & roi
+            if new_box.area > 0:
+                new_indexes[i].insert(j, tuple(new_box), hit.object)
+                j += 1
+
+    new_datasets = []
+    for index in new_indexes:
+        ds = deepcopy(dataset)
+        ds.index = index
+        new_datasets.append(ds)
+
+    return new_datasets
+
+
+def time_series_split(
+    dataset: GeoDataset, lengths: Sequence[Union[int, float, Tuple[int, int]]]
+) -> List[GeoDataset]:
+    """Split a GeoDataset on its time dimension to create non-overlapping GeoDatasets.
+
+    Args:
+        dataset: dataset to be split
+        lengths: lengths, fractions or pairs of timestamps (start, end) of splits
+            to be produced
+
+    Returns
+        A list of the subset datasets.
+
+    .. versionadded:: 0.5
+    """
+    minx, maxx, miny, maxy, mint, maxt = dataset.bounds
+
+    totalt = maxt - mint
+
+    if not all(isinstance(x, tuple) for x in lengths):
+        if not (sum(lengths) == 1 or sum(lengths) == totalt):  # type: ignore[arg-type]
+            raise ValueError(
+                "Sum of input lengths must equal 1 or the dataset's time length."
+            )
+
+        if any(n <= 0 for n in lengths):  # type: ignore[operator]
+            raise ValueError("All items in input lengths must be greater than 0.")
+
+        if sum(lengths) == 1:  # type: ignore[arg-type]
+            lengths = [totalt * f for f in lengths]  # type: ignore[operator]
+
+        lengths = [
+            (mint + offset - length, mint + offset)  # type: ignore[operator, misc]
+            for offset, length in zip(accumulate(lengths), lengths)
+        ]
+
+    new_indexes = [
+        Index(interleaved=False, properties=Property(dimension=3)) for _ in lengths
+    ]
+
+    _totalt = 0
+    for i, (start, end) in enumerate(lengths):  # type: ignore[misc]
+        if start >= end:
+            raise ValueError(
+                "Pairs of timestamps in lengths must have end greater than start."
+            )
+
+        if start < mint or end > maxt:
+            raise ValueError(
+                "Pairs of timestamps in lengths can't be out of dataset's time bounds."
+            )
+
+        if any(  # type: ignore[misc]
+            start < x < end or start < y < end for x, y in lengths[i + 1 :]
+        ):
+            raise ValueError("Pairs of timestamps in lengths can't overlap.")
+
+        # remove one second from each BoundingBox's maxt to avoid overlapping
```

Comment:
In `torchgeo.datasets.utils.disambiguate_timestamp`, we remove 1 microsecond. Not sure if this really matters though, as long as overlap is avoided. I doubt anyone will be doing anything with satellite imagery at the millisecond resolution.

Reply:
I'll change it to microsecond for consistency